lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -51,13 +51,21 @@ class DescribeClass < Base
           )
         PATTERN
 
-        def on_top_level_describe(node, (described_value, _))
-          return if shared_group?(root_node)
-          return if valid_describe?(node)
-          return if describe_with_rails_metadata?(node)
-          return if string_constant_describe?(described_value)
+        def_node_matcher :described, <<~PATTERN
+          (block $(send #{RSPEC} :describe $_ ...) ...)
+        PATTERN
+
+        def on_top_level_group(top_level_node)
+          node, described = described(top_level_node)
+
+          return unless described
```

Comment:
Can we have `!nil?` in the matcher and skip this check

Reply:
I keep forgetting about the yielding feature of a node pattern.

This comment of yours, however, led to a major refactoring 😆